### DISCUSSION TITLE
Adding support for A-EQDSK files?

### DISCUSSION THREAD

**Discussion by hansec** [OpenFUSIONToolkit/OpenFUSIONToolkit#28] [Polls]
Periodically we have gotten requests to include support for writing EQDSK "A" files. References for the content of these files can be found in [OMFIT](https://omfit.io/_modules/omfit_classes/omfit_eqdsk.html#OMFITaeqdsk.add_aeqdsk_documentation) and [here](https://nstx.pppl.gov/nstx/Software/Applications/a-g-file-variables.txt).

The content of these files includes a hodgepodge of information, some of which is machine specific and/or easily extracted from an EQDSK "G" file. In addition there is a lot of information that is not widely used any more (eg. "Shafranov boundary line integral parameter") or well defined (eg. "plasma minor radius in cm"). As a result, many codes other than EFIT set lots of these quantities to default values that have no meaning.

So I want to solicit community input on the value of these files and their use before we go deep to add support for this output. Below is a simple yes/no style poll on desire for this feature. In addition to answering this poll, please provide the following information on your use case for these files:

1. What are the files used for?
   * If they are used as input to another code, what is the name of that code?
2. What fields in the file are actually used?
3. Does your workflow need any other EQDSK files (eg. "K", "M", etc.) needed?
   * If so, please answer the first two questions for each of those files as well.

**Comment by JaeBeom1019** [comment]
Certainly, I understand that GEQDSK files are typically utilized in code inputs. However, it's unfortunate that the GEQDSK file, which serves as the output file of OFT, doesn't encompass all of OFT's solutions, notably crucial plasma parameters like coil current. When disseminating OFT's solutions to others, I advocate for the implementation of AEQDSK to include supplementary information alongside the profiles in GEQDSK.

Presently, sharing a solution necessitates the additional transmission of supplementary information alongside the GEQDSK file containing the profile data. With the introduction of AEQDSK, all pertinent parameters could be transmitted simultaneously , streamlining the process for both the sender and the recipient.(much more convinient) This would obviate the need for repeated data exchange, enhancing convenience and efficiency in data exchange.

For the last(3.) case, I don’t need other EQDSK files except GEQDSK and AEQDSK file in my workflow. However if other EQDSKs becomes available, I would be grateful as it would be used in a more extensive way.
